docs/proposals/edge-node-upgrade.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Edge Node Upgrade
+authors:
+- "@gy95"
+  approvers:
+  creation-date: 2022-06-26
+  last-updated: 2022-06-26
+  status: implementable
+---
+
+# Edge Node Upgrade
+
+## Motivation
+
+Edge node upgrade management is a key feature required for upgrading edge nodes from remote cloud side in edge computing.
+This proposal addresses how can we upgrade edge nodes from the cloud, and synchronize the edge node upgrade result status between edge nodes and cloud.
+
+### Goals
+
+Edge node upgrade management must:
+* provide APIs for upgrading edge nodes from the cloud.
+* synchronize the edge node upgrade result between cloud and edge nodes.
+
+## Proposal
+We propose using Kubernetes [Custom Resource Definitions (CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+to describe Upgrade metadata/status and a controller to synchronize Upgrade between edge and cloud.
+
+### Use Cases
+
+* Describe Upgrade properties.
+  * Users can describe Upgrade properties and access mechanisms to interact with / control the Upgrade.
+* Perform CRUD operations on Upgrade from cloud.
+  * Users can create, update and delete Upgrade metadata from the cloud via the CRD APIs exposed by the Kubernetes API server.
+* Report Upgrade properties values.
+  * Edge nodes can report upgrade result status.
+
+
+## Upgrade Controller Design
+The Upgrade controller starts two separate goroutines called  `upstream` controller and `downstream` controller.
+These are not separate controllers as such but named here for clarity.
+The job of the downstream controller is to synchronize the Upgrade updates from the cloud to the edge node.
+The job of the upstream controller is the reverse.
+
+
+## Synchronizing Upgrade
+
+The below illustrations describe the flow of events that would occur when Upgrade property values are updated from the cloud/edge.
+
+- Users create/modify the Upgrade CR directly using `kubectl`.
```

Comment:
I want to know what field would be allowed to update  after we create the CR?

Reply:
And if the CR already completed, can we edit the cr?

Reply:
The CR looks like a `job` , we should consider all above

Reply:
add Upgrade related webhook to do these restriction